src/main/java/org/mqttbee/mqtt/handler/connect/MqttConnectHandler.java

```diff
@@ -117,14 +130,15 @@ private void addClientData(@NotNull final Channel channel) {
      */
     private void writeConnect(@NotNull final ChannelHandlerContext ctx) {
         ctx.writeAndFlush(connect).addListener(future -> {
+            connectSent = true;
```

Comment:
connectSent must be set to true directly after the if (!connectSent ... check
otherwise it is possible to send the connect twice as the listener is invoked asynchronously

Reply:
done. renamed it to connectCalled because we do not really know if it was sent at this point.